### ISSUE TITLE
[Feature Request]: Toggle switch/button

### ISSUE THREAD

**Issue by PWagner1** [Krypton-Suite/Standard-Toolkit#1116]
It would be nice to have a toggle switch as part of the toolkit

![download](https://github.com/Krypton-Suite/Standard-Toolkit/assets/949607/dc951b22-196f-4fb3-91d8-24a8525c03c8)

**Comment by Smurf-IV** [comment]
I thought this was already in the extended kit ??
If not, then that is where it should be ?
Or has winforms now got it in for .net8 ?

**Comment by Wagnerp** [comment]
> I thought this was already in the extended kit ??

It is, but I wondered if it could be a variant of the KCheckButton? The toolkit was originally designed when the Windows OS did not have a toggle switch controls. These were introduced with Windows 8 and are still being used.

**Comment by Smurf-IV** [comment]
> The toolkit was originally designed when the Windows OS did not have a toggle switch controls. These were introduced with Windows 8 and are still being used.

I understand that lot's of controls are added as the OS moves on into different recommended app styles (`UWP` etc)
But I believe the Standard kit is to replace the the Controls currently in the Visual Studio Winform Toolkit.
And Extended "mainly" to bring in the newer controls from UWP etc.
And Also the extended to enhance / extend the standard kit.

What would you like to change the philosophy to be starting with V8.1 onwards ? 
If its documented then Devs will know what to expect.

**Comment by Smurf-IV** [comment]
Also, As per #1062; When VS has added [Other Controls](https://github.com/dotnet/winforms/blob/main/docs/roadmap.md#add-and-improve-wrappers-for-missing-win32-controls) to VS, then they can also be moved / implemented in the standard kit. A bit like what has been done to retrofit the new button in MessageBox [Cancel / retry / continue] down to the other frameworks.

**Comment by Wagnerp** [comment]
> Also, As per #1062; When VS has added [Other Controls](https://github.com/dotnet/winforms/blob/main/docs/roadmap.md#add-and-improve-wrappers-for-missing-win32-controls) to VS, then they can also be moved / implemented in the standard kit. A bit like what has been done to retrofit the new button in MessageBox [Cancel / retry / continue] down to the other frameworks.

Might be worth keeping an eye out

**Comment by Smurf-IV** [comment]
OK, Will **also** need
- [ ] Demo app
- [ ] Palette Designer addition(s)
- [ ] List of other controls that where also introduced for >= Win 8

**Comment by PWagner1** [comment]
@Smurf-IV & @giduac 

Is it possible to create it based off a KCheckBox?

**Comment by Smurf-IV** [comment]
> @Smurf-IV & @giduac
> 
> Is it possible to create it based off a KCheckBox?

Yes, as the actions needed by the "in-designer" and events would be the same.
Drawing and Palette designer additions would need to be done (Unless colours could be borrowed from button highlights etc)

**Comment by PWagner1** [comment]
> > @Smurf-IV & @giduac
> > Is it possible to create it based off a KCheckBox?
> 
> Yes, as the actions needed by the "in-designer" and events would be the same. Drawing and Palette designer additions would need to be done (Unless colours could be borrowed from button highlights etc)

Also, we would need to facilitate labels for 'On/Off'

**Comment by giduac** [comment]
@PWagner1  & @Smurf-IV 

Have a look at the Winform checkbox using the flatstyle properties.
You can achieve as similar effect and use images for the checked (on/off) state.

That might be done in KCB with the underlying winforms checkbox.

That could be a sperate KControl or integrated into the KCBox...

**Comment by giduac** [comment]
Or use the KCheckbutton with images...

**Comment by Smurf-IV** [comment]
> Or use the KCheckbutton with images...

But then you would not get the "Slide" action when toggling ;-(

**Comment by giduac** [comment]
@Smurf-IV 
Depends...
If you animate the toggle from the point where the check state changes and displaying several images in a row until the on/of state is reached....

If you added some kind of the above to a kcheckbutton....
Or use that as a base...

Another thing is the scaling of the imagery when the user resizes the button.
If possible it should be a vector image

**Comment by PWagner1** [comment]
@Smurf-IV & @giduac 

Thoughts? (still need to add text)

https://github.com/user-attachments/assets/8a790548-3ff5-426d-b60a-e5618b984045

**Comment by giduac** [comment]
Do you have better video, this one looks a bit fuzzy?

**Comment by PWagner1** [comment]
> Do you have better video, this one looks a bit fuzzy?

@giduac 

Some improvements 

https://github.com/user-attachments/assets/90c95e81-c6e4-4f52-82dc-a890d302cb00

**Comment by giduac** [comment]
Push the branch please, then we can have a go?

**Comment by PWagner1** [comment]
> Push the branch please, then we can have a go?

@giduac 

Go ahead, I'm not done yet... :)

**Comment by Smurf-IV** [comment]
> A better look

In the last video it does not appear to "Slide"
This is the main feature of the "Toggle buttons"
And also the colours change for slider buttons to indicate On / off states, where are these done ?
I'm not sure this is ready as completed yet.

**Comment by PWagner1** [comment]
> > A better look
> 
> In the last video it does not appear to "Slide" This is the main feature of the "Toggle buttons" And also the colours change for slider buttons to indicate On / off states, where are these done ? I'm not sure this is ready as completed yet.

It'll be done in stages. Colours need to be implemented. For the first version possibly the text colours will change & then go from there...

**Comment by nathancorvussolis** [comment]
Thank you for creating great software.

I found a problem while using Krypton.Toolkit 105.26.4.110 from NuGet.
NullReferenceException is raised at KryptonToggleSwitch.OnGlobalPaletteChanged() when the palette is changed.
Member variable `_paletteRedirect` is maybe always null.

https://github.com/Krypton-Suite/Standard-Toolkit/blob/536b030ace75c8122b601cd0ade37b2135bd9762/Source/Krypton%20Components/Krypton.Toolkit/Controls%20Toolkit/KryptonToggleSwitch.cs#L40

https://github.com/Krypton-Suite/Standard-Toolkit/blob/536b030ace75c8122b601cd0ade37b2135bd9762/Source/Krypton%20Components/Krypton.Toolkit/Controls%20Toolkit/KryptonToggleSwitch.cs#L693-L713
